src/Bridge/Doctrine/Orm/ItemDataProvider.php

```diff
@@ -112,7 +115,7 @@ private function addWhereForIdentifiers(array $identifiers, QueryBuilder $queryB
 
             $queryBuilder->andWhere($expression);
 
-            $queryBuilder->setParameter($placeholder, $value);
+            $queryBuilder->setParameter($placeholder, $value, $classMetadata->getTypeOfField($identifier));
```

Comment:
Oh, this is good, I didn't knew you can add a type as third argument! Maybe we can still add this fix to the 2.1 branch!